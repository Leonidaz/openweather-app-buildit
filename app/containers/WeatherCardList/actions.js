/*
 *
 * WeatherCardList actions
 *
 */

import * as types from './constants';

export const fetchWeatherData5 = () => ({
  type: types.FETCH_WEATHER5_REQUESTED,
});
export const receiveWeather5Success = response => ({
  type: types.FETCH_WEATHER5_SUCCESS,
  response,
});
export const receiveWeather5Failure = error => ({
  type: types.FETCH_WEATHER5_FAILURE,
  error,
});
