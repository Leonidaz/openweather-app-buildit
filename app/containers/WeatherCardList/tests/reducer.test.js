
import { fromJS } from 'immutable';
import { weatherData } from 'api/tests/mockWeatherData';
import weatherCardListReducer from '../reducer';
import * as actions from '../actions';

const mockedData = fromJS(weatherData);

describe('weatherCardListReducer', () => {
  it('should return the initial state', () => {
    expect(weatherCardListReducer(undefined, {})).toMatchSnapshot();
  });

  it('should populate the store with mocked data', () => {
    expect(weatherCardListReducer(undefined, actions.receiveWeather5Success(mockedData))).toEqual(mockedData);
  });
});
