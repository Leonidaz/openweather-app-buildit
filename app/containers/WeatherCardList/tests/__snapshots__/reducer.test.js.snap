exports[`weatherCardListReducer should return the initial state 1`] = `Object {}`;
